app/controllers/courses_controller.rb

```diff
@@ -160,26 +160,26 @@ def load_trainees
   end
 
   def load_course_subjects
-    @course_subjects = @course.course_subjects.includes(subject: :tasks).paginate page: params[:page],
-      per_page: Settings.pagination.per_page
+    @course_subjects = @course.course_subjects.includes(subject: :tasks)
+                              .paginate page: params[:page],
+                                per_page: Settings.pagination.per_page
   end
 
   def transaction_update_course
     ActiveRecord::Base.transaction do
       @course.close!
       @course.user_courses.update_all status: :finished
-      @course.course_subjects.update_all status: :finished
-      @course.course_subjects.each do |course_subject|
-        course_subject.user_subjects.update_all status: :finished
-        course_subject.user_subjects.each do |user_subject|
-          user_subject.user_tasks.update_all status: :finished
-        end
-      end
+      course_subjects = @course.course_subjects
+      course_subjects.update_all status: :finished
+      user_subjects = UserSubject.by_course_subjects course_subjects.ids
+      user_subjects.update_all status: :finished
```

Comment:
bữa nói dùng callback ở model cậu chưa dùng hì

Reply:
Cái này nữa demo anh Huy góp ý đó chị. Callback nó cũng tương tự như vòng lặp for, nó chạy từng cái where 1. Anh Huy góp ý dùng where IN như này. Dù có bao nhiêu row thì cũng chỉ 4 câu lệnh SQL trên là update hết thôi.